File: src/lambda/dms-switch/utils/get-dms-task.ts
import { CloudFormationClient } from '@aws-sdk/client-cloudformation';
import { listStackResources } from './list-stack-resources';

export const getDmsTask = async ({ cf, StackName }: {
  cf: CloudFormationClient;
  StackName: string;
}): Promise<string> => {
  const resources = await listStackResources({ cf, StackName, resources: [] });
  const dmsTask = resources.filter((res) => res.ResourceType === "AWS::DMS::ReplicationTask")[0].PhysicalResourceId;
  return `${dmsTask}`;
};

File: src/lambda/dms-switch-py/utils/get_dms_task.py
from utils.list_stack_resources import list_stack_resources


def get_dms_task(cf, stack_name):
    resources = list_stack_resources(cf, stack_name, [])
    dms_tasks = [res['PhysicalResourceId']
                 for res in resources if res['ResourceType'] == 'AWS::DMS::ReplicationTask']
    if dms_tasks:
        dms_task = dms_tasks[0]
        return dms_task
    else:
        return None
